src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,28 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error while adding movie", e);
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Movie movie;
+        try (Session session = factory.openSession()) {
+            movie = session.get(Movie.class, id);
+        } catch (Exception e) {
+            throw new DataProcessingException("Error while getting movie with id: " + id, e);
```

Comment:
Similar to the add method, the exception message should include details about the movie that failed to be retrieved. Include the `id` in the exception message to make the error more informative.